### ISSUE TITLE
Weird behavior with plurals

### ISSUE THREAD

**Issue by MushuPlatypus** [i18next/react-i18next#1053]
**Describe the bug**
I have a language file with this content:
{
  "PETS-PLURAL-STRING_0": "{{n}} pets",
  "PETS-PLURAL-STRING_1": "{{n}} pet",
  "PETS-PLURAL-STRING_2": "{{n}} pets",
}

When using the useTranslation hook like this:
{t('PETS-PLURAL-STRING_0')}
{t('PETS-PLURAL-STRING_1')}
{t('PETS-PLURAL-STRING_2')}

I get: pets pet pets.

When doing this:
{t('PETS-PLURAL-STRING', { count: 0 })}
{t('PETS-PLURAL-STRING', { count: 1 })}
{t('PETS-PLURAL-STRING', { count: 2 })}

I get: pet pets pet.

The output from the count comes from PETS-PLURAL-STRING_1 PETS-PLURAL-STRING_0 PETS-PLURAL-STRING_1 

I would expect to output: pets pet pets.

I am initing i18next like this:
  i18n.init({
    load: 'currentOnly',
    fallbackLng: 'en',
    simplifyPluralSuffix: false,
    pluralSeparator: '_',
    keySeparator: '.',
    nonExplicitWhitelist: true,
    backend: {
      // for all available options read the backend's repository readme file
      loadPath: '/locales/{{lng}}.json',
      fetch,
    },
    interpolation: {
      escapeValue: false,
    },
    initImmediate: false,
    debug: process.env.NODE_ENV !== 'production',
    preload: false,
    react: {
      wait: true,
      useSuspense: true,
    },
  })

**Occurs in react-i18next version**
11.3.1

**Comment by adrai** [comment]
English has just a singular and a plural:
https://www.i18next.com/translation-function/plurals
Here you can try: https://jsfiddle.net/sm9wgLze
Or did I miss something?

**Comment by jamuhl** [comment]
@adrai is right...even with non-simplified plural naming there are only two forms -> _0, _1

**Comment by MushuPlatypus** [comment]
Even if it only uses 0 and 1 it reverses them:
{t('PETS-PLURAL-STRING', { count: 0 })} should return 'pets'.  (_0)
{t('PETS-PLURAL-STRING', { count: 1 })} should return 'pet'.   (_1)
{t('PETS-PLURAL-STRING', { count: 2 })} should return 'pets'.  (_0)

Instead now it returns:
'pet' (_1), 'pets' (_0) and 'pet' (_1).

**Comment by jamuhl** [comment]
because the `en` rule is https://github.com/i18next/i18next/blob/master/src/PluralResolver.js#L43

-> all but count = 1 gives true which is `1`

-> 1 gives `_0` (ONE;  ['ONE', 'OTHER'], index 0)
-> 0, 2, 3, 4, 5, 6, ... gives _1 (OTHER;  ['ONE', 'OTHER'], index 1)

**Comment by MushuPlatypus** [comment]
How would I use the count value then?  I have to support 27 languages, some of which have some complicated plurals.

In our old solution we could just slap count in and it would work perfectly.  Of course this was with i18next version 10.0.3 so a lot could have happened.

**Comment by jamuhl** [comment]
Not sure where you your struggle come from? There is no difference in usage. Not using that `simplifyPluralSuffix: false,` you would be able to use `key, key_plural` for most languages.

There was a short time we used some numbers representing the pluralform: https://www.i18next.com/misc/json-format#i-18-next-json-v2 as this made not much sense we changed to use the indexed like in gettext or cldr pluralforms

Details here: https://www.i18next.com/translation-function/plurals

Test tool: https://jsfiddle.net/sm9wgLze

**Comment by adrai** [comment]
tldr:
either you do it this way:
```json
{
  "PETS-PLURAL-STRING": "{{count}} pet",
  "PETS-PLURAL-STRING_plural": "{{count}} pets"
}
```

or this way:
```json
{
  "PETS-PLURAL-STRING_0": "{{count}} pet",
  "PETS-PLURAL-STRING_1": "{{count}} pets"
}
```

**Comment by MushuPlatypus** [comment]
I think that is what I'm doing, but right now it I add 22 pets to my basket I get this output: 22 pet.

If I add one pet I get this: 1 pets.

According to the documentation the json should be like this:
{
  "key_0": "zero",      (pets)
  "key_1": "singular",  (pet)
  "key_2": "two",       (pets)
  "key_3": "few",
  "key_4": "many",
  "key_5": "other"
}

i18next.t('key', {count: 0}); // -> "zero"
i18next.t('key', {count: 1}); // -> "singular"
i18next.t('key', {count: 2}); // -> "two"

**Comment by adrai** [comment]
See my example above. _0 is singular _1 is plural

**Comment by jamuhl** [comment]
> According to the documentation the json should be like this:
> {
> "key_0": "zero", (pets)
> "key_1": "singular", (pet)
> "key_2": "two", (pets)
> "key_3": "few",
> "key_4": "many",
> "key_5": "other"
> }

that's for arabic...not for english

**Comment by MushuPlatypus** [comment]
I guess I'll have our editors set to work moving the translations around then, so it will work with the updated version :)
